src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrNodeResource.java

```diff
@@ -182,7 +183,10 @@ private InputStream getInputStream() {
             try {
                 // find the content node: for nt:file it is jcr:content
                 // otherwise it is the node of this resource
-                Node content = node.isNodeType(NT_FILE)
+                Node content = (node.isNodeType(NT_FILE) ||
+                                (node.isNodeType(JcrConstants.NT_FROZENNODE) &&
+                                 node.hasProperty(JcrConstants.JCR_FROZENPRIMARYTYPE) &&
```

Comment:
I think this is not necessary. `jcr:frozenPrimaryType` is mandatory and autocreated for `nt:frozenNode` https://docs.adobe.com/docs/en/spec/jcr/2.0/3_Repository_Model.html#3.13.4.1%20nt:frozenNode